lib/capybara/poltergeist/inspector.rb

```diff
@@ -22,9 +22,12 @@ def url
       "//localhost:#{port}/"
     end
 
-    def open
+    def open(scheme = nil)
       if browser
-        Process.spawn(browser, url)
+        scheme = 'http' unless scheme == 'https' # scheme could be 'file' or 'about' or something else
```

Comment:
@twalpole I'm not quite sure what you're suggesting for an alternative, as there is no way to prevent a caller from passing any argument they want. The intent here is for the caller to request an inspector connection that uses the given scheme. If the requested scheme is `http` or `https` then that request will be honored, but otherwise, since there's no other way to connect to the server, HTTP is used and the request ignored. I don't see any value in imposing the burden of checking what the scheme is on the caller, nor is it the caller's responsibility to know what schemes are valid for connecting to the inspector, nor is it helpful to the caller to know what scheme is actually being used for the connection.

Furthermore, this leaves the method open for future enhancement, so someone could possibly enable the `file` scheme for connecting to the inspector using a UNIX named pipe, for example, without requiring changes to client code.